gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CaResources.java

```diff
@@ -53,113 +62,93 @@ public CaResources(ILogger logger) {
         this.logger = logger;
     }
 
-    public Optional<AliasContentCert> readPalantirRootCaFromSystemTruststore() {
-        return systemCertificates().flatMap(CaResources::selectPalantirCertificate);
+    public static Optional<AliasContentCert> readPalantirRootCaFromSystemTruststore() {
+        return selectPalantirCertificate(systemCertificates());
     }
 
-    public Optional<AliasContentCert> maybeGetCertificateFromSerialNumber(String serialNumber, String alias) {
-        return systemCertificates()
-                .map(certs -> selectCertificates(certs, Map.of(serialNumber, alias)))
-                .orElseGet(Stream::empty)
-                .findFirst();
+    public void importAllSystemCerts(Path jdkInstallationDirectory) {
+        importCertificates(jdkInstallationDirectory, parseCerts(systemCertificates()));
     }
 
-    public void importCertInJdk(AliasContentCert aliasContentCert, Path jdkInstallationDirectory) {
-        Os os = CurrentOs.get();
-        switch (os) {
-            case MACOS:
-            case LINUX_GLIBC:
-            case LINUX_MUSL:
-                unixImportCertInJdk(aliasContentCert, jdkInstallationDirectory);
-                logger.log(String.format(
-                        "Successfully imported CA certificate %s into the JDK truststore",
-                        aliasContentCert.getAlias()));
-                break;
-            case WINDOWS:
-                logger.logError(
-                        String.format("Importing certificates for OS type '%s' is not yet supported", os.uiName()));
-                break;
-        }
-    }
-
-    private void unixImportCertInJdk(AliasContentCert aliasContentCert, Path jdkInstallationDirectory) {
+    private void importCertificates(Path jdkInstallationDirectory, List<Certificate> certificates) {
         try {
-            if (isCertificateInTruststore(jdkInstallationDirectory, aliasContentCert.getAlias())) {
-                logger.log("Certificate already exists in the truststore, skipping...");
-                return;
+            char[] passwd = "changeit".toCharArray();
+            Path jksPath = jdkInstallationDirectory.resolve("lib/security/cacerts");
+            KeyStore jks = loadKeystore(passwd, jksPath);
+            Set<BigInteger> existingSerialNumbers = getExistingCertificates(jks);
+            logger.log(String.format("Certificate %s imported successfully into the KeyStore.", existingSerialNumbers));
+            List<X509Certificate> newCertificates = certificates.stream()
+                    .map(X509Certificate.class::cast)
+                    .filter(certificate -> !existingSerialNumbers.contains(certificate.getSerialNumber()))
+                    .collect(Collectors.toList());
+            for (X509Certificate certificate : newCertificates) {
+                String alias = getAlias(certificate);
+                logger.log(String.format("Certificate %s imported successfully into the KeyStore.", alias));
```

Comment:
I just tried this branch on my laptop, and I got some interesting logging. First, I got a big long line of numbers, which I don't know what it is:

![Screenshot 2024-09-06 at 13 31 35](https://github.com/user-attachments/assets/131dd396-0054-4fee-a757-7ae881f2194c)

I also ended up with multiple certs being imported with the same alias:

```
Certificate GradleJdks_computerName imported successfully into the KeyStore.
Certificate GradleJdks_computerName imported successfully into the KeyStore.
```

I think these are being overwritten. We can probably fix this by putting the serial number into the alias too?

Reply:
yes, I think I added the wrong log line here: https://github.com/palantir/gradle-jdks/pull/403/files/4b6a87a4346fae20fb06dbbd67bc740850013f74#diff-f0333751b6ca40fe3d118c9f2fce18e089d40d62e5bb0649a7a0635b82e5df56R79. Deleting it